lib/beaker/host_prebuilt_steps.rb

```diff
@@ -289,7 +292,13 @@ def get_ip(host)
     # @param [Host] host the host to act upon
     # @param [String] etc_hosts The string to append to the /etc/hosts file
     def set_etc_hosts(host, etc_hosts)
-      host.exec(Command.new("echo '#{etc_hosts}' > /etc/hosts"))
+      if host['platform'] =~ /freebsd/
+        # FreeBSD gets weird about special characters, we have to go a little OTT here
+        freebsd_hosts = etc_hosts.gsub(/\t/,'\\t').gsub(/\n/,'\\n')
+        host.exec(Command.new("printf '#{freebsd_hosts}' > /etc/hosts"))
```

Comment:
Might be good here to have a host method like host.echo_msg(msg), which can then be overridden for freebsd hosts.